waspc/packages/ts-inspect/src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { getExportsOfFiles } from "./exports.js";
+
+interface ExportsRequest {
+  filenames: string[],
+  tsconfig?: string,
+}
+
+async function readStdin(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    let chunks = '';
+    process.stdin.on('data', (data) => {
+      chunks += data;
+    });
+    process.stdin.on('end', () => resolve(chunks));
+    process.stdin.on('close', () => resolve(chunks));
+    process.stdin.on('error', (err) => reject(err));
+  });
+}
+
+async function main() {
+  const inputStr = await readStdin();
+  const input = JSON.parse(inputStr);
+  if (!Array.isArray(input)) {
+    throw new Error('Expected JSON array from stdin');
+  }
+  const requests: ExportsRequest[] = input;
```

Comment:
I would recommend defining a schema with Zod and parsing the input so you can be sure about the format 100% https://zod.dev/?id=basic-usage